src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -1327,6 +1327,27 @@ public void Add(IEnumerable<T> items)
         }
 
         /// <summary>
+        /// Special add for TypeTable type entries that removes redundant file entries.
+        /// </summary>
+        internal void AddTypeTableTypesInfo(IEnumerable<T> items)
+        {
+            if (!(items is IEnumerable<SessionStateTypeEntry>)) { throw new PSInvalidOperationException(); }
+
+            lock (_syncObject)
+            {
+                foreach (var element in items)
+                {
+                    var typeEntry = element as SessionStateTypeEntry;
+                    if (typeEntry.TypeData != null)
+                    {
+                        // Skip type file entries.
```

Comment:
I would be surprised if we have tests covering the scenario - I think you'll need to write some new tests.